lib/ansible/runner.rb

```diff
@@ -0,0 +1,29 @@
+module Ansible
+  class Runner
+    class << self
+      def run(env_vars, extra_vars, playbook_path)
+        run_via_cli(env_vars, extra_vars, playbook_path)
+      end
+
+      private
+
+      def run_via_cli(env_vars, extra_vars, playbook_path)
+        result = %x(#{format_env_vars(env_vars)} #{ansible_command.shellescape} #{format_extra_vars(extra_vars)} #{playbook_path.to_s.shellescape})
+        JSON.parse(result)
+      end
+
+      def ansible_command
+        # TODO add possibility to use custom path, e.g. from virtualenv
+        "ansible-playbook"
+      end
+
+      def format_env_vars(env_vars)
+        env_vars.map { |key, value| "#{key.shellescape}='#{value.shellescape}'" }.join(" ")
+      end
+
+      def format_extra_vars(extra_vars)
+        "--extra-vars '#{JSON.dump(extra_vars)}'"
```

Comment:
I think you need to shell_escape the JSON, right?  what if it has a single quote in it?